.github/workflows/lint.yaml

```diff
@@ -1,49 +1,25 @@
 ---
 name: "Lint"
-on: # yamllint disable-line rule:truthy
+on:  # yamllint disable-line rule:truthy
   push:
     branches:
       - "main"
   pull_request:
     branches: ["*"]
 jobs:
   lint:
-    name: "Format & Lint"
+    # Uses pre-commit to run all checks
     runs-on: "ubuntu-latest"
-    strategy:
-      matrix:
-        python-version:
-          - "3.10"
-      fail-fast: false
     steps:
       - uses: "actions/checkout@v4"
-      - uses: "bewuethr/yamllint-action@v1.3.0"
-        with:
-          config-file: ".yamllint"
+      - name: "Install poetry"
+        run: "pipx install poetry"
       - uses: "actions/setup-python@v5"
```

Comment:
This makes deps available for pyright, and should also do a better job of caching deps between runs.